### PR TITLE
Fix profile api in swagger

### PR DESCRIPTION
**Pull request by wisererik**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*:  #1017 

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: openapi-spec/swagger.yaml [hunk 1]
```diff
@@ -1905,6 +1902,31 @@ components:
...
+      in: query
+      description: Start position of the pagination query
+      schema:
+        minimum: 0
+        type: integer
+        format: int32
+    sortDir:
+      name: sortDir
+      in: query
+      description: Asending or descending
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]:
```suggestion
      description: Ascending or descending
```

File: openapi-spec/swagger.yaml [hunk 2]
```diff
@@ -1997,78 +2052,104 @@ components:
...
           format: int64
         provisioningPolicy:
+          default: thick
+          description: The provisioning policy of the pool. It can be thick or thin.
           type: string
           enum:
             - Thin
             - Thick
         compression:
+          description: The optimization policy to remove redundant data within the file
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]: Compression is not removing redundant data but reduction in the number of bits in the data to save storage

File: openapi-spec/swagger.yaml [hunk 3]
```diff
@@ -1997,78 +2052,104 @@ components:
...
+      description: DataStorageLoS can be used to describe a service option covering
+        storage provisioning and availability.
     IOConnectivityLoS:
-      description: >-
-        IOConnectivityLoS can be used to specify the characteristics of storage
-        connectivity.
       type: object
       properties:
         accessProtocol:
+          description: The type of access protocol. This can be iscsi, rbd, fibre channel, or DSWare(Data Service Middleware), nvmeof and nfs
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]: Are we supporting DSWare?

File: openapi-spec/swagger.yaml [hunk 4]
```diff
@@ -1997,78 +2052,104 @@ components:
...
             - NFS
         maxIOPS:
+          example: 150
+          description: The maximum IOPS(Input/Output Operations Per Second) supported by storage. For e.g. 150IOPS
           type: integer
           format: int64
-        maxBWS:
+        minIOPS:
+          example: 50
+          description: The minimum IOPS(Input/Output Operations Per Second) supported by storage.In otherwords, performance can not dropped below this level. For e.g. 50IOPS
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]:
```suggestion
          description: The minimum IOPS(Input/Output Operations Per Second) supported by storage.In other words, performance can not drop below this level. For e.g. 50IOPS
```

File: openapi-spec/swagger.yaml [hunk 5]
```diff
@@ -1997,78 +2052,104 @@ components:
...
           type: integer
           format: int64
-        minIOPS:
+        maxBWS:
+          description: The maximum bandwidth suported by storage. For e.g. 5mbps
+          example: 5
           type: integer
           format: int64
         minBWS:
+          description: The minimum bandwidth suported by storage. In otherwords, performance can not dropped below this level. For e.g. 1mbps
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]:
```suggestion
          description: The minimum bandwidth supported by storage. In other words, performance can not drop below this level. For e.g. 1mbps
```

File: openapi-spec/swagger.yaml [hunk 6]
```diff
@@ -1997,78 +2052,104 @@ components:
...
+          default: false
+          example: true
+          description: Isolated storage is designed to prevent data corruption and data loss
           type: boolean
         minLifetime:
+          description: It is an ISO 8601 duration that specifies the minimum lifetime required for the replica
           type: string
           example: P3Y6M4DT12H30M5S
         RecoveryGeographicObjective:
+          description: It is supported failure domain from where data can be restored after disaster
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]:
```suggestion
          description: Supported failure domain from where data can be restored after disaster
```

File: openapi-spec/swagger.yaml [hunk 7]
```diff
@@ -2078,22 +2159,30 @@ components:
             - Row
             - Server
         RecoveryPointObjectiveTime:
+          description: It's an ISO 8601 time interval that specifies the maximum source information that may be lost on failure. In the case of isIsolated = false, failure of the domain is not considered.
           type: string
           example: P3Y6M4DT12H30M5S
         RecoveryTimeObjective:
+          description: It specifies the relative time required to make a replica available as a source
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]: Is it relative?
```suggestion
          description: It specifies the time required to make a replica available as a source
```

File: openapi-spec/swagger.yaml [hunk 8]
```diff
@@ -2078,22 +2159,30 @@ components:
...
+          example: offline
           type: string
           enum:
             - OnlineActive
             - OnlinePassive
             - Nearline
             - Offline
         ReplicaType:
+          description: The value specify the type of repilca
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]:
```suggestion
          description: This value specify the type of replica
```

File: openapi-spec/swagger.yaml [hunk 9]
```diff
@@ -2214,27 +2303,68 @@ components:
             key2: false
             key3:
               key31: value31
-    ProfileSpec:
-      description: >-
-        An OpenSDS profile is identified by a unique name and ID. With adding
-        extra properties, each profile can contains a set of tags of storage
+    ProfileCreateSpec:
+      description: An OpenSDS profile is identified by a unique name and ID. With
+        adding extra properties, each profile can contains a set of tags of storage
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]:
```suggestion
        adding extra properties, each profile can contain a set of tags of storage
```

File: openapi-spec/swagger.yaml [hunk 10]
```diff
@@ -2214,27 +2303,68 @@ components:
...
+          $ref: '#/components/schemas/ReplicationPropertiesSpec'
+        snapshotProperties:
+          $ref: '#/components/schemas/SnapshotPropertiesSpec'
+        dataProtectionProperties:
+          $ref: '#/components/schemas/DataProtectionPropertiesSpec'
+        customProperties:
+          $ref: '#/components/schemas/CustomPropertiesSpec'
+    ProfileRespSpec:
+      description: An OpenSDS profile is identified by a unique name and ID. With
+        adding extra properties, each profile can contains a set of tags of storage
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]:
```suggestion
        adding extra properties, each profile can contain a set of tags of storage
```

File: openapi-spec/swagger.yaml [hunk 11]
```diff
@@ -2431,34 +2569,42 @@ components:
...
     CustomPropertiesSpec:
-      description: >-
-        CustomPropertiesSpec is a map of keys and JSON object that represents
-        the customized properties of profile, such as requested capabilities
-        including diskType, latency, deduplication, compression and so forth.
       type: object
       additionalProperties: {}
+      description: CustomPropertiesSpec is a map of keys and JSON object that represents
+        the customized properties of profile, such as requested capabilities including
+        diskType, latency, deduplicaiton, compression and so forth.
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]:
```suggestion
        disk type, latency, de-duplication, compression and so forth.
```

File: openapi-spec/swagger.yaml (hunk 3 above)
Comment (wisererik) [reply openapi-spec/swagger.yaml]: yes, it's based on huawei storage

File: openapi-spec/swagger.yaml (hunk 7 above)
Comment (wisererik) [reply openapi-spec/swagger.yaml]: yes, it's from swordfish

File: openapi-spec/swagger.yaml (hunk 1 above)
Comment (wisererik) [reply openapi-spec/swagger.yaml]: done

File: openapi-spec/swagger.yaml (hunk 2 above)
Comment (wisererik) [reply openapi-spec/swagger.yaml]: done

File: openapi-spec/swagger.yaml (hunk 4 above)
Comment (wisererik) [reply openapi-spec/swagger.yaml]: done

File: openapi-spec/swagger.yaml (hunk 5 above)
Comment (wisererik) [reply openapi-spec/swagger.yaml]: done

File: openapi-spec/swagger.yaml (hunk 6 above)
Comment (wisererik) [reply openapi-spec/swagger.yaml]: done

File: openapi-spec/swagger.yaml (hunk 8 above)
Comment (wisererik) [reply openapi-spec/swagger.yaml]: done

File: openapi-spec/swagger.yaml (hunk 9 above)
Comment (wisererik) [reply openapi-spec/swagger.yaml]: done

File: openapi-spec/swagger.yaml (hunk 10 above)
Comment (wisererik) [reply openapi-spec/swagger.yaml]: done

File: openapi-spec/swagger.yaml (hunk 11 above)
Comment (wisererik) [reply openapi-spec/swagger.yaml]: done

**Pull request merged**